### ISSUE TITLE
Demand reconfirmation of decision if action is changed the last moment before acting

### ISSUE THREAD

**Issue by louwers** [pokerth/pokerth#330]
This happens quite often to me and I noticed it happens other players as well: when confirming an action out of turn (e.g. 'check') it is possible that just before you click, the action changes (e.g. into 'call $5000'). Then unintentionally that action is performed.

Other poker platforms handle this problem by requiring a reconfirmation if you clicked just before the action changed (perhaps require reconfirmation if an action was made within 0.5 seconds after the action has changed).

**Comment by doitux** [comment]
Do you have the "accidentally call blocker" activated like in the screenshot?
![screenshot_20170507_170223](https://cloud.githubusercontent.com/assets/1719554/25782242/2c36ee00-3347-11e7-9ec4-fc6c5fa0670d.png)

**Comment by louwers** [comment]
I just checked, yes I have that enabled.

**Comment by q4z1** [comment]
You really mean the moment just before you click the action, when the player before you decides to make a bet for example?
The option mentioned above does not handle that indeed.
We might put it on a feature request list.

**Comment by louwers** [comment]
I think so. 
(about to click call) -> raise (0.2 seconds pass) -> click

**Comment by doitux** [comment]
https://github.com/pokerth/pokerth/blob/bc3e7166817e90b92c8dce53c9474930ef32307a/src/gui/qt/gametable/gametableimpl.cpp#L1794

I thought this should block this ... may be you can check from this point?

**Comment by louwers** [comment]
@doitux I found the issue. The buttons work (i.e. you can't call within a second), but if you call/raise using the function keys, it will still come through.

**Comment by louwers** [comment]
@q4z1 Please re-label as a bug if you agree with my assessment by looking at the code or if you can replicate it.

**Comment by albmed** [comment]
@louwers, thanks for your effort. However, what you mean by 
> buttons work, but .... the function keys, it will still come through. 

You mean, function keys don't wait one second, but clicking with mouse does? Can you clarify, please?

**Comment by louwers** [comment]
@albmed If you're click an action button when it's not your turn yet, it is possible that just before you click the button, the action changes into one you did not intend to make. To prevent this from happening, there is an option that allows you to disable the button for a second after the action that button represents has changed.

Try clicking the 'call' button just after someone bet, and you'll find that it doesn't work.

If you use the function keys to call, for example, even if someone has just made a bet, it will still go through. The protection does not work.

This is the clearest I can explain it. The best way is just trying to call a bet using the buttons and using the function keys to see that, indeed, the button is blocked, while the function keys work when they shouldn't.

**Comment by albmed** [comment]
Thanks @louwers, I'll check it out. 
I thought it was controlled in `MyActionButton::event()` function, in which if `eatMyEvents` was `true`, it didn't performed the check action (both both from keyboard or mouse). I'm probably wrong...  ;)

**Comment by q4z1** [comment]
albmed and me checked code - and it looks like the disable should be active for FKey presses too. We will do some tests next days.

**Comment by albmed** [comment]
As @doitux commented, perhaps the best option would be to overwrite `click()` slot in `MyActionButton`. One possible implementation could be (to be tested): 

In **myactionbutton.cpp**: 

```C
void MyActionButton::click()
{
    if (!eatMyEvents) {
        QPushButton::click();
    }
}
```

In **myactionbutton.h**: 

```C
class MyActionButton : public QPushButton {
//[....] 
private slots: 
   void click();
}
```

**Comment by q4z1** [comment]
Issue solved - will be available in upcoming 1.1.2 release.
